File: app/src/main/java/com/reitler/got/model/statistics/MatchStatistics.java
package com.reitler.got.model.statistics;

import java.util.ArrayList;
import java.util.List;
import java.util.stream.Collectors;

public class MatchStatistics {

    private List<MatchStatistic> matches = new ArrayList<>();

    public int getCount(){
        return matches.size();
    }

    public double getAvgScoresPerMatch(){
        double absoluteScores = matches.stream().collect(Collectors.summingDouble(MatchStatistic::getScoreSum));
        return absoluteScores / getCount();
    }

    public double getAvgScoresPerPlayer(){
        double sumAvgScores = matches.stream().collect(Collectors.summingDouble(MatchStatistic::getAvgPlayerScore));
        return sumAvgScores
                / getCount();
    }

    public void addMatch(MatchStatistic match){
        this.matches.add(match);
    }
}

File: app/src/main/java/com/reitler/got/model/statistics/MatchStatistics.kt
package com.reitler.got.model.statistics

import java.util.*
import java.util.stream.Collectors

class MatchStatistics {
    private val matches: MutableList<MatchStatistic> = ArrayList()
    val count: Int
        get() = matches.size
    val avgScoresPerMatch: Double
        get() {
            if (count == 0) {
                return 0.0
            }
            val absoluteScores = matches.stream().collect(Collectors.summingDouble { m -> m.scoreSum.toDouble() })
            return absoluteScores / count
        }
    val avgScoresPerPlayer: Double
        get() {
            if (count == 0) {
                return 0.0
            }
            val sumAvgScores = matches.stream().collect(Collectors.summingDouble(MatchStatistic::avgPlayerScore))
            return (sumAvgScores
                    / count)
        }
    val totalRemains: Int
        get() = if(matches.isEmpty()) 0 else matches.sumBy { it.scoreSum }
    val maxRemains: Int
        get() = if(matches.isEmpty()) 0 else matches.maxOf { it.maxScore }

    fun addMatch(match: MatchStatistic) {
        matches.add(match)
    }
}